Description: Upstream changes by gniibe, fixing FTBFS on sparc and ia64
Author: NIIBE Yutaka <gniibe@fsij.org>
Last-Update: 2020-08-25
Bug-Debian: http://bugs.debian.org/489844
Forwarded: https://github.com/uim/sigscheme/issues/10

Index: sigscheme-0.8.5/test-c/test-gc-protect-coll.c
===================================================================
--- sigscheme-0.8.5.orig/test-c/test-gc-protect-coll.c	2011-02-15 16:55:00.000000000 +0900
+++ sigscheme-0.8.5/test-c/test-gc-protect-coll.c	2011-02-15 16:55:02.000000000 +0900
@@ -192,10 +192,10 @@
     for (i = 0; i < N_OBJS; i++)
         TST_TN_TRUE(scm_gc_protectedp(auto_objs[i]));
 
-#if TRY_TESTS_THAT_PASS_IN_MOST_CASES
     /* unprotect again */
     for (i = 0; i < N_OBJS; i++)
         scm_gc_unprotect(&auto_objs[i]);
+#if TRY_TESTS_THAT_PASS_IN_MOST_CASES
     for (i = 0; i < N_OBJS; i++)
         TST_TN_FALSE(scm_gc_protectedp(auto_objs[i]));
 #endif
@@ -227,7 +227,9 @@
 
 TST_CASE(tst_6, "GC indirect protection via on-heap object reference")
 {
+#if TRY_TESTS_THAT_PASS_IN_MOST_CASES
     ScmObj lst;  /* unprotected */
+#endif
 
     TST_TN_FALSE(scm_gc_protected_contextp());
 
